mitiq/cdr/clifford_training_data.py

```diff
@@ -0,0 +1,516 @@
+                method_select,
+                method_replace,
+            )
+
+        circuits_list.append(projected_circuit)
+        # this information is to make sure the probabilistic methods are
+        #  working as expected:
+        angles_original_list.append(angles_original)
+        angles_replaced_list.append(angles_replaced)
+    return circuits_list, angles_original_list, angles_replaced_list
+
+
+def _map_to_near_clifford(
+    circuit: Circuit,
+    fraction_non_clifford: float,
+    random_state: np.random.RandomState,
+    method_select: str = "random",
+    method_replace: str = "closest",
+    **additional_options: dict,
+) -> Tuple[Circuit, List[float], List[float]]:
+    """Function to take the information in some circuit of interest and
+        return a near-Clifford circuit as constructed according to
+        some user defined methods.
+    Args:
+        circuit: cirq.Circuit object of origional circuit of interest.
+        fraction_non_clifford: the fraction of non-Clifford gates to replace
+                               in the circuit of interest.
+        method_select: string defining the method used to select the
+                       non-Clifford gates to replace: 'random' or
+                       'probabilistic'.
+        method_replace: string defining method used to replace selected
+                        non-Clifford gates: 'closest', 'random' or
+                        'probabilistic'.
+        random_state: Seed for sampling.
+        sigma_select: width of probability distribution used in selection
+                      of non-Clifford gates to replace, only has effect if
+                      method_select = 'probabilistic'
+        sigma_replace: width of probability distribution used in replacement
+                       of selected non-Clifford gates, only has effect if
+                       method_replace = 'probabilistic'.
+        Returns:
+            Circuit: Near-Clifford projected circuit.
+            List[float]: list of angles replaced that were replaced in the
+                         training circuit.
+            List[float]: list of angles that were inserted in the training
+                         circuit.
+    Raises:
+        Exception: If additional options does not contain one or two keys
+                   'sigma_select' and/or 'sigma_replace' both equal to some
+                   float.
+        .
+    """
+    # set the seed for sampling, for replacement and selection:
+    (random_state_select, random_state_replace) = random_state.randint(
+        10 ** (8), size=2
+    )
+    random_state_select = np.random.RandomState(random_state_select)
+    random_state_replace = np.random.RandomState(random_state_replace)
+    # get the operations from the circuit and find the non-cliff angles:
+    operations = np.array(list(circuit.all_operations()))
+    positions = np.linspace(1, len(operations), len(operations))
+    gates = _get_gates(operations)
```

Comment:
```suggestion
    gates = np.array([op.gate for op in circuit.all_operations()])
```

Reply:
And remove `operations` above, I don't think you use this anywhere except to get the gates. If so you can remove `_get_gates` below.